### DISCUSSION TITLE
Template Element Layout does not support custom module paths

### DISCUSSION THREAD

**Discussion by BenParizek** [craftcms/cms#10714] [Ideas] [answered]
I'm trying to build a few Element layouts using the new unified Element Editor in Craft 4. I've created my default fields as FieldLayoutElements and they are displaying fine. But I also have the need to output some additional HTML that isn't a field. I think I'd do this with something like the Template UI Element `craft\fieldlayoutelements\Template` however if I try to pass the Template UI Element a path from my module it fails to find the module.

I can create a simple UI Element that does what I need but I think it would be nice if either:

1. The Template UI Element allowed us to pass in paths from our custom modules (my module is a Yii module not a Craft plugin if that matters). Right now it just appears to be looking in the Craft template path and I haven't had luck swapping the Template Path (but I'm open to being told I'm missing something)

2. The Template UI Element or another simple HTML UI Element let us just render the template we needed ahead of time and pass the rendered $html to the UI Element.

The settings I've been exploring this in are via `Element::metaFieldsHtml()` and the `FieldLayout::EVENT_CREATE_FORM` event. One of the scenarios I'm trying to support is my Reports plugin which allows custom Data Sources to provide an arbitrary settings HTML blob for the Element page, so I'm effectively letting an integration add custom fields via HTML to my Element form so it's a little bit different than the default custom fields scenario on Elements and more like how Fields allow custom fields to add settings. I'm just mentioning this for context. I have another scenario where I used to output a cheat sheet of rules in the sidebar but now that the sidebar HTML is more strict it may make more sense to output this on a secondary tab or in the main field layout as HTML, so I have two scenarios that may benefit from this.

Everything else is working fine and the data gets submitted correctly if I create a simple HtmlUiElement Field Layout Element. 

But, as I've run into this twice already in my first pass at updating Elements I thought I'd post as it feels like it would be nicer if the simple HTML use case was supported by Craft, which I think the Template UI Element is intended to do minus the path restrictions.

**Comment by brandonkelly** [comment, marked as answer]
Just [added](https://github.com/craftcms/cms/commit/f97c298b89e6d6efbae8a1ecc17c678901e0bc96) a new `craft\fieldlayoutelements\Html` field layout element class, which you can use in field layouts that aren’t meant to be user-editable, rather than `Template`. Pass whatever HTML you want into its constructor, which could originate from a template if you want:

```php
new Html($html)
```

**Reply by brandonkelly** [reply]
Craft 4.0.0-beta.2 is out now with that addition.
